security/s2a/internal/crypter/counter_test.go

```diff
@@ -0,0 +1,109 @@
+package crypter
+
+import (
+	"math"
+	"testing"
+)
+
+type counterTest struct {
+	desc                     string
+	counter, expectedCounter uint64
+	overflow                 bool
+}
+
+func TestCounterInc(t *testing.T) {
+	for _, test := range []counterTest{
+		{
+			desc:            "basic 1",
+			counter:         0,
+			expectedCounter: 1,
+		},
+		{
+			desc:            "basic 2",
+			counter:         123,
+			expectedCounter: 124,
+		},
+		{
+			desc:            "almost overflow",
+			counter:         math.MaxUint64 - 1,
+			expectedCounter: math.MaxUint64,
+		},
+		{
+			desc:     "max overflow",
+			counter:  math.MaxUint64,
+			overflow: true,
+		},
+	} {
+		t.Run(test.desc, func(t *testing.T) {
+			// Make first getAndIncrement call. This should return the same value
```

Comment:
This could already be the case, but please make sure this line is not longer than 80 chars.